bundles/org.openhab.binding.jablotron/src/main/java/org/openhab/binding/jablotron/internal/discovery/JablotronDiscoveryService.java

```diff
@@ -0,0 +1,171 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.jablotron.internal.discovery;
+
+import static org.openhab.binding.jablotron.JablotronBindingConstants.*;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.discovery.*;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerService;
+import org.openhab.binding.jablotron.internal.handler.JablotronBridgeHandler;
+import org.openhab.binding.jablotron.internal.model.JablotronDiscoveredService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link JablotronDiscoveryService} is responsible for the thing discovery
+ * process.
+ *
+ * @author Ondrej Pecta - Initial contribution
+ */
+@NonNullByDefault
+public class JablotronDiscoveryService extends AbstractDiscoveryService
+        implements DiscoveryService, ThingHandlerService {
+    private final Logger logger = LoggerFactory.getLogger(JablotronDiscoveryService.class);
+
+    private @Nullable JablotronBridgeHandler bridgeHandler;
+
+    private @Nullable ScheduledFuture<?> discoveryJob = null;
+
+    public JablotronDiscoveryService() {
+        super(DISCOVERY_TIMEOUT_SEC);
+        logger.debug("Creating discovery service");
+    }
+
+    private void startDiscovery() {
+        if (this.bridgeHandler.getThing().getStatus() == ThingStatus.ONLINE) {
```

Comment:
As `bridgeHandler` is Nullable, you have to do a null check before accessing it. You can store it to a local variable and do the null check on that. Same for the others in this file.

Reply:
fixed